source/pki_utils.c

```diff
@@ -226,3 +226,76 @@ int aws_read_and_decode_pem_file_to_buffer_list(
 
     return AWS_OP_SUCCESS;
 }
+
+int aws_decode_pem_to_der_buf(
+    struct aws_allocator *alloc,
+    const struct aws_byte_cursor *pem_cursor,
+    struct aws_byte_buf *der_buf) {
+    aws_byte_buf_clean_up(der_buf);
+    struct aws_array_list base_64_buffer_list;
+
+    if (aws_array_list_init_dynamic(&base_64_buffer_list, alloc, 2, sizeof(struct aws_byte_buf))) {

```

Comment:
nit: having init here is not ideal, especially since s_convert_pem_to_raw_base64 knows how many elements that array should have before adding to it

Reply:
The function is deprecated.

Reply:
I revisited the algorithm in `aws_decode_pem_to_buffer_list`, and I didn't see a good way to work around this, as `s_convert_pem_to_raw_base64` determines the buffer list size, while it requires an initialized array list.